pypit/arproc.py

```diff
@@ -22,195 +23,95 @@
-        fitfunc_model = np.interp(xball, xbarr[argsrt], fitfunc)
-        bgmod += arcyproc.background_model(fitfunc_model, xapix, yapix, sciframe.shape[0], sciframe.shape[1])
-#		np.save("bspl/xbarr_ord{0:d}".format(o+1),xbarr)
-#		np.save("bspl/ybarr_ord{0:d}".format(o+1),ybarr)
-#		np.save("bspl/ebarr_ord{0:d}".format(o+1),ebarr)
-#		print min(np.min(xbarr),xstr), max(np.max(xbarr),xfin), ncoeff, k
-#		np.save("bspl/pixlocn_ord{0:d}".format(o+1),slf._pixlocn)
-#		np.save("bspl/tilts_ord{0:d}".format(o+1),slf._tilts[:,o])
-#		np.save("bspl/xmod_ord{0:d}".format(o+1),xmod)
-#		np.save("bspl/ycen_ord{0:d}".format(o+1),ycen[:,o])
-#		np.save("bspl/xpix_ord{0:d}".format(o+1),xpix)
-#		np.save("bspl/ypix_ord{0:d}".format(o+1),ypix)
-#		print "saved all!"
-#		#mod_yarr = cybspline.bspline_fit(xmod, xbarr, ybarr, ebarr, min(np.min(xbarr),xstr), max(np.max(xbarr),xfin), ncoeff, k)
-#		bgmod += arcyutils.bspline_fitmod(xbarr, ybarr, ebarr, min(np.min(xbarr),xstr), max(np.max(xbarr),xfin), ncoeff, k, slf._pixlocn, slf._tilts[:,o], xmod, ycen[:,o], xpix, ypix)
-
-    arutils.ds9plot(bgmod.astype(np.float))
-    arutils.ds9plot((sciframe-bgmod).astype(np.float))
-
-    #exsci, exerr = arcyextract.extract_weighted(frame, error, badpixmask, pixcen[:,o], piycen[:,o], slf._pixlocn, ordtilt[:,o], ordxcen[:,o], ordycen[:,o], ordwid[:,o], ordwnum[o], ordlen[o], ordnum[o], argf_interpnum)
-
-# 	ordcen = 0.5*(slf._lordloc + slf._rordloc)
-# 	ordwid = np.ceil(np.median(np.abs(slf._lordloc - slf._rordloc),axis=0)).astype(np.int)/2
-# 	cord_pix = artrace.phys_to_pix(ordcen, slf._pixlocn, 1)
-# 	print cord_pix
-# 	print ordwid
-# 	test_rect = arcyextract.rectify_fast(sciframe, cord_pix, ordwid, -999999.9)
-# 	arutils.ds9plot(test_rect)
-# 	for o in range(norders):
-# 		pass
-
-
-    assert(False)
-    # Mask out ordpix pixels where there is target flux
-    ordpix = None
-    # Prepare and fit the sky background pixels in every order
-    msgs.work("Multiprocess this step to make it faster")
-    skybg = np.zeros_like(sciframe)
-    for o in range(norders):
-        xpix, ypix = np.where(ordpix==1+o)
-        msgs.info("Preparing sky pixels in order {0:d}/{1:d} for a b-spline fit".format(o+1,norders))
-        #xbarr, ybarr = cybspline.prepare_bsplfit(arc, pixmap, tilts, xmod, ycen, xpix, ypix)
-        msgs.info("Performing b-spline fir to oversampled sky background in order {0:d}/{1:d}".format(o+1,norders))
-        #ncoeff, k = flt.shape[0], 1
-        #mod_yarr = cybspline.bspline_fit(xmod, xbarr, ybarr, ebarr, min(np.min(xbarr),xstr), max(np.max(xbarr),xfin), ncoeff, k)
-        #skybg += cybspline.bspline_fitmod(xbarr, ybarr, ebarr, min(np.min(xbarr),xstr), max(np.max(xbarr),xfin), ncoeff, k, pixmap, tilts, xmod, ycen, xpix, ypix)
-
-    # Subtract the background
-    msgs.info("Subtracting the sky background from the science frame")
-    return sciframe-skybg, skybg
+    word = np.where((slf._slitpix[det - 1] == slitn + 1) & (slf._scimask[det - 1] == 0))
```

Comment:
Done - I should say, I wrote this faster background subtraction routine because our other `bg_subtraction` routine is too time consuming for multiple slits (or orders). This routine also has the advantage of not using cython, which we are starting to phase out.